app/org/sagebionetworks/bridge/dynamodb/DynamoCriteriaDao.java

```
@@ -0,0 +1,77 @@
+ */
+@Component
+public class DynamoCriteriaDao implements CriteriaDao {
+
+    private DynamoDBMapper criteriaMapper;
+    
+    @Resource(name = "criteriaMapper")
+    final void setCriteriaMapper(DynamoDBMapper criteriaMapper) {
+        this.criteriaMapper = criteriaMapper;
+    }
+    
+    // Until criteria as an interface is shifted off of SchedulePlan/Subpopulation, 
+    // we must copy from an interface to an implementation.
+    @Override
+    public Criteria copyCriteria(String key, Criteria criteria) {
+        Criteria actualCriteria = Criteria.create();
+        actualCriteria.setKey(key);
+        if (criteria != null) {
+            actualCriteria.setMinAppVersion(criteria.getMinAppVersion());
+            actualCriteria.setMaxAppVersion(criteria.getMaxAppVersion());
+            actualCriteria.setAllOfGroups(criteria.getAllOfGroups());
+            actualCriteria.setNoneOfGroups(criteria.getNoneOfGroups());
+        }
+        return actualCriteria;        
+    }
+    
+    @Override
+    public void createOrUpdateCriteria(Criteria criteria) {
+        checkNotNull(criteria);
+        checkArgument(isNotBlank(criteria.getKey()));
+
+        Criteria actualCriteria = copyCriteria(criteria.getKey(), criteria);
+        criteriaMapper.save(actualCriteria);
+    }
+    
+    @Override
+    public Criteria getCriteria(String key) {
+        checkArgument(isNotBlank(key));
+        
+        DynamoCriteria hashKey = new DynamoCriteria();
+        hashKey.setKey(key);
+
+        return criteriaMapper.load(hashKey);
+    }
+
+    @Override
+    public void deleteCriteria(String key) {
+        checkArgument(isNotBlank(key));
+        
+        DynamoCriteria hashKey = new DynamoCriteria();
+        hashKey.setKey(key);
+        
+        Criteria criteria = criteriaMapper.load(hashKey);
+        if (criteria != null) {
+            criteriaMapper.delete(hashKey);
+        }
```

Do we want to error if the key doesn't exist? Or silently return? Should we add a unit test case for this?